### PR TITLE
Add geometry feature

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Example code demonstrates how to write shape information from an OFF file to an `NXoff_geometry` group in a NeXus file. It also shows how to output entire instrument geometry, comprising `NXoff_geometry`, `NXcylindrical_geometry` and `NXtransformations` as a single OFF file which can then be opened in CAD software.

A file passes as compatible with the feature if it has at least one `NXoff_geometry` or `NXcylindrical_geometry` group containing the expected datasets for that `NXclass`.

Includes example file `example_nx_geometry.nxs`

*To Test:*
You can test by adding these lines to `8CB1EBAE3B2DA51D/recipe.py` and running it:
```python
if __name__ == "__main__":
    nx_file = h5py.File("../../../examples/example_nx_geometry.nxs", "r")
    example = NeXusOFF(nx_file["raw_data_1"])
    example.output_shape_to_off_file("example.off")
```
I've been using [Geomview](http://www.geomview.org/download/) to render OFF files. The download page contains OSX build instructions and rpm and deb packages, or on Ubuntu you can just `apt install geomview`.

### CODE REVIEW

**Comment by matthew-d-jones** [comment]
@zjttoefs Example code now demonstrates outputting complete geometry to an OFF file; I've updated the PR description.

File: src/recipes/8CB1EBAE3B2DA51D/recipe.py [hunk 1]
```diff
@@ -0,0 +1,669 @@
+import h5py
+import numpy as np
+
+
+class GeometryExamples:
```
Comment (zjttoefs) [inline src/recipes/8CB1EBAE3B2DA51D/recipe.py]: The name of this class is not ideal. "Examples" makes you suspect it doesn't parse the file. Why not OFFGeometry or something simple like that?

**Review by zjttoefs — COMMENTED** [review]
Am I reading this correctly that the feature would extract all OFF shapes and merge them with the correct respective orientation and distance? How does it deal with scanned dimensions?

**Comment by zjttoefs** [comment]
Looked a bit more in detail. I've pushed a commit with a class name that's at least somewhat better. I've also changed the title. No need for "example". 
Still need to confirm it actually merges multiple OFF buckets correctly (and if so under which circumstances).

**Pull request merged**